lib/vdsm/storage/lvm.py

```diff
@@ -910,29 +916,39 @@ def getPvs(self, vgName):
         return pvs
 
     def getVg(self, vgName):
-        # Get specific VG
+        """
+        Get specific VG.
+        Raise a VolumeGroupDoesNotExist for LVM command errors or missing VG.
+        """
         vg = self._vgs.get(vgName)
         if not vg or vg.is_stale():
             self.stats.miss()
             vgs = self._reload_single_vg(vgName)
+            if vgName not in vgs:
+                raise se.VolumeGroupDoesNotExist(vg_name=vgName)
             vg = vgs.get(vgName)
         else:
             self.stats.hit()
         return vg
 
     def getVgs(self, vgNames):
-        """Reloads all the VGs of the set.
+        """
+        Reload all the VGs of the set.
 
         Can block for suspended devices.
-        Fills the cache but not uses it.
-        Only returns found VGs.
+        Fill the cache but do not use it.
+        Return only found VGs.
```

Comment:
Good changes, but not needed for this PR. Lets remove every change we don't need now to minimize this PR.